### PR TITLE
DEPR: Index.__getitem__[bool] GH#44051

### LINKED ISSUE #44051: API: Index.__getitem__(True) returns 2D ndarray
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
Not clear if this is intentional:


import numpy as np
import pandas as pd

arr = np.random.randn(4)
idx = pd.Index(arr)

idx[True]

>>> idx[True]
array([[ 1.39365849, -0.03655461,  0.17112114, -1.4612337 ]])
```

### Issue Description

For other cases where indexing on index._data returns 2D (e.g. `idx[:, None]`) we issue a FutureWarning, but this goes through a different code path.

### Expected Behavior

I don't have an opinion here, just think whatever we do should be intentional

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #44051
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/common.py [hunk 1]
```diff
@@ -670,6 +670,21 @@ def test_getitem_2d_deprecated(self, simple_index):
 
         assert isinstance(res, np.ndarray), type(res)
 
+        if not isinstance(idx, RangeIndex):
```
Comment (jreback) [inline pandas/tests/indexes/common.py:673]: what about a list of bools?

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -540,6 +540,7 @@ Other Deprecations
 - Deprecated parameter ``names`` in :meth:`Index.copy` (:issue:`44916`)
 - A deprecation warning is now shown for :meth:`DataFrame.to_latex` indicating the arguments signature may change and emulate more the arguments to :meth:`.Styler.to_latex` in future versions (:issue:`44411`)
 - Deprecated :meth:`Categorical.replace`, use :meth:`Series.replace` instead (:issue:`44929`)
+- Deprecated :meth:`Index.__getitem__` with a bool key; use ``index.values[key]`` to get the old behavior (:issue:`44051`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:543]: do we have other deprecated indexing like this? can you co-locate (followup ok too)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexes/common.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/common.py:673]: not sure, but definitely out of scope

File: doc/source/whatsnew/v1.4.0.rst (hunk 2 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.4.0.rst:543]: i think from a previous version, so the note will be in a different file

File: pandas/tests/indexes/common.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/indexes/common.py:673]: kk

File: doc/source/whatsnew/v1.4.0.rst (hunk 2 above)
Comment (jreback) [reply doc/source/whatsnew/v1.4.0.rst:543]: kk that's fine then

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks the first comment pls followup when you have a chance (list of bools when indexing)

**Pull request merged**